sympy/core/mul.py

```diff
@@ -549,7 +549,11 @@ def _handle_for_oo(c_part, coeff_sign):
 
         # 0
         elif coeff is S.Zero:
-            # we know for sure the result will be 0
+            # we know for sure the result will be 0 except the multiplicand
+            # is infinity
+            if(len(c_part) > 0):
+                if c_part[0].is_commutative and c_part[0].is_finite == False:
```

Comment:
`c_part[0].is_infinite` - probably that you want instead.

Perhaps just:
```python
if any(c.is_infinite for c in c_part):
    return ...
```

FYI: c_part IS a commutative part...  Oh.